### PR TITLE
Trigger Failover On 3092 And 1022

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Include samples if adding new API, and include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ResourceThrottleRetryPolicy.cs [hunk 1]
```diff
@@ -133,6 +163,8 @@ private object GetExceptionMessage(Exception exception)
         /// <param name="request">The request being sent to the service.</param>
         public void OnBeforeSendRequest(DocumentServiceRequest request)
         {
+            this.isMultiMasterWriteRegion = !request.IsReadOnlyRequest
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/ResourceThrottleRetryPolicy.cs:166]: I think you are missing the condition that >1 write region need to be available (honoring preferred regions, excludedRegions etc.) - Could be included in GlobalEndpointManager.CanUseMultipleriteRegion but glancing at the code I did not see this check.

File: Microsoft.Azure.Cosmos/src/direct/GoneAndRetryWithRequestRetryPolicy.cs [hunk 2]
```diff
@@ -428,6 +445,13 @@ private static bool IsPartitionKeyRangeGone(TResponse response, Exception except
                 || (response?.StatusCode == HttpStatusCode.Gone && response?.SubStatusCode == SubStatusCodes.PartitionKeyRangeGone);
         }
 
+        private static bool IsGoneWithLeaseNotFound(TResponse response, Exception exception)
+        {
+            return exception is LeaseLostException
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/direct/GoneAndRetryWithRequestRetryPolicy.cs:450]: No LeaseLostException has completely different meaning - that is thrown when Change Feed Processor is losing the CFP lease.

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
Looks good except for few minor comments

File: Microsoft.Azure.Cosmos/src/direct/GoneAndRetryWithRequestRetryPolicy.cs (hunk 2 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/direct/GoneAndRetryWithRequestRetryPolicy.cs:450]: Yes, this is taken care in the original msdata PR: `1453704`

Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/direct/GoneAndRetryWithRequestRetryPolicy.cs:450]: I have created a new exception type: `LeaseNotFoundException`. That PR has more context on it. We can follow up on that PR itself!

**Comment by kundadebdatta** [comment]
Was a demo PR. Closing since the original ones got merged.

**Pull request closed without merging**